src/snowflake/cli/plugins/git/commands.py

```diff
@@ -38,6 +42,79 @@ def _repo_path_argument_callback(path):
 )
 
 
+def _assure_repository_does_not_exist(repository_name: str) -> None:
+    om = ObjectManager()
+    try:
+        om.describe(
+            object_type=ObjectType.GIT_REPOSITORY.value.cli_name,
+            name=repository_name,
+        )
+        raise ClickException(f"Repository '{repository_name}' already exists")
+    except ProgrammingError:
+        pass
+
+
+@app.command("setup", requires_connection=True)
+def setup(
+    repository_name: str = RepoNameArgument,
+    **options,
+) -> CommandResult:
+    """
+    Sets up a git repository object.
+
+    You will be prompted for:
+
+    * url - address of repository to be used for git clone operation
+
+    * secret - Snowflake secret containing authentication credentials. Not needed if origin repository does not require
+    authentication for RO operations (clone, fetch)
+
+    * API integration - object allowing Snowflake to interact with git repository.
+    """
+    _assure_repository_does_not_exist(repository_name)
+    manager = GitManager()
+
+    url = typer.prompt("Origin url")
+
+    secret = None
+    secret_needed = typer.confirm("Use secret for authentication?")
+    if secret_needed:
+        use_existing_secret = typer.confirm("Use existing secret?")
+        if use_existing_secret:
+            secret = typer.prompt("Secret identifier")
+        else:
+            cli_console.step("Creating new secret")
+            secret = f"{repository_name}_secret"
+            username = typer.prompt("username")
+            password = typer.prompt("password/token", hide_input=True)
+            manager.create_password_secret(
```

Comment:
I would move object creation to the end, in this way there's a clear separation of "prompts" and "actions" WDYT?

Reply:
done